std/range/package.d

```diff
@@ -1228,6 +1231,382 @@ unittest
+                    }
+                }
+
+                static if (allSameType && allSatisfy!(hasAssignableElements, R))
+                {
+                    @property void back(RvalueElementType v)
+                    {
+                        foreach (i, Unused; R)
+                        {
+                            if (i == index)
+                            {
+                                source.fields[i].back = v;
+                                return;
+                            }
+                        }
+                        assert(0);
+                    }
+                }
+            }
+
+            static if (allSatisfy!(hasLength, R))
+            {
+                @property size_t length()
+                {
+                    foreach (i, Unused; R)
+                    {
+                        if (i == index)
+                            return source.fields[i].length;
+                    }
+                    assert(0);
+                }
+
+                alias opDollar = length;
+            }
+
+            static if (allSatisfy!(isRandomAccessRange, R))
+            {
+                auto ref opIndex(size_t index)
+                {
+                    foreach (i, Range; R)
+                    {
+                        if (i == this.index)
+                        {
+                            static if (isInfinite!(Range))
+                            {
+                                return source.fields[i][index];
+                            }
+                            else
+                            {
+                                return fixRef(source.fields[i][index]);
+                            }
+                        }
+                    }
+                    assert(0);
+                }
+
+                static if (allSatisfy!(hasMobileElements, R))
+                {
+                    RvalueElementType moveAt(size_t index)
+                    {
+                        foreach (i, Range; R)
+                        {
+                            if (i == this.index)
+                            {
+                                return .moveAt(source.fields[i], index);
+                            }
+                        }
+                        assert(0);
+                    }
+                }
+
+                static if (allSameType && allSatisfy!(hasAssignableElements, R))
+                    void opIndexAssign(ElementType v, size_t index)
+                    {
+                        foreach (i, Range; R)
+                        {
+                            if (i == this.index)
+                            {
+                                source.fields[i][index] = v;
+                                return;
+                            }
+                        }
+                        assert(0);
+                    }
+            }
+
+            static if (allSatisfy!(hasLength, R) && allSatisfy!(hasSlicing, R))
```

Comment:
```d
void main()
{
    import std.range: choose;
    int[] a;
    long[] b;
    auto c = choose(0, a, b);
}
```

That should compile and `c` should not have an `opSlice`. At the moment, it fails compilation with `std/range/package.d(1532): Error: mismatched function return type inference of long[] and int[]`.